dataprofiler/labelers/column_name_model.py

```diff
@@ -0,0 +1,253 @@
+        check_values_dict,
+        positive_threshold,
+        include_label,
+        show_confidences,
+    ):
+        """Calculate similarity scores for positive examples."""
+        scores = self._model(
+            list_of_column_names,
+            check_values_dict,
+            self._make_lower_case,
+            rapidfuzz.fuzz.token_sort_ratio,
+            include_label=include_label,
+        )
+
+        output_dictionary = {}
+        for i in range(len(list_of_column_names)):
+            if scores[i][0] > positive_threshold:
+                output_dictionary[list_of_column_names[i]] = {}
+                output_dictionary[list_of_column_names[i]]["pred"] = check_values_dict[
+                    scores[i][1]
+                ]["label"]
+                if show_confidences:
+                    output_dictionary[list_of_column_names[i]]["conf"] = scores[i][0]
+
+        return output_dictionary
+
+    def _construct_model(self):
+        pass
+
+    def _reconstruct_model(self):
+        pass
+
+    def _need_to_reconstruct_model(self):
+        pass
+
+    def reset_weights(self):
+        """Reset weights function."""
+        pass
+
+    @require_module(["rapidfuzz"])
+    def _model(
+        self,
+        list_of_column_names,
+        check_values_dict,
+        processor,
+        scorer,
+        include_label=False,
+    ):
+        scores = []
+
+        check_values_list = [dict["attribute"] for dict in check_values_dict]
+
+        model_outputs = rapidfuzz.process.cdist(
+            list_of_column_names, check_values_list, processor=processor, scorer=scorer
+        )
+
+        for iter_value, ngram_match_results in enumerate(model_outputs):
+            column_result = [np.max(ngram_match_results)]
+            if include_label:
+                index_max_result = ngram_match_results.argmax(axis=0)
+                column_result.append(index_max_result)
+            scores.append(column_result)
+
+        return scores
+
+    def predict(
+        self,
+        data,
+        batch_size=None,
+        show_confidences=False,
+        verbose=True,
+        include_label=True,
+    ):
+        """
+        Apply the `process.cdist` for similarity score on input list of strings.
+
+        :param data: list of strings to predict upon
+        :type data: iterator
+        :param batch_size: does not impact this model and should be fixed to not
+            be required.
+        :type batch_size: N/A
+        :param show_confidences: whether user wants prediction confidences
+        :type show_confidences:
+        :param verbose: Flag to determine whether to print status or not
+        :type verbose: bool
+        :return: char level predictions and confidences
+        :rtype: dict
+        """
+        false_positive_dict = self._parameters["false_positive_dict"]
+        if false_positive_dict:
+            data = self._compare_negative(
+                data, false_positive_dict, negative_threshold=50
```

Comment:
is this `negative_threshold` value arbitrary? Should it be a constant at the top of the file or in a config somewhere?

Reply:
good catch -- nah this should be a param and not hard coded